File: backend/src/main/java/com/twitter/demo/like/service/LikeService.java
package com.twitter.demo.like.service;

import com.twitter.demo.like.dto.LikeCreateRequest;
import com.twitter.demo.like.dto.LikeResponse;
import com.twitter.demo.like.dto.LikedUserDTO;

import java.util.List;

public interface LikeService {
    LikeResponse likeTweet(LikeCreateRequest request);

    void dislikeTweet(LikeCreateRequest request);

    void deleteAllByUserId(Long userId);


    /*
    Long countByTweetId(Long tweetId);

    List<LikedUserDTO> getLikedUsers(Long tweetId);

     */
}

File: twitter_frontend/twitter-frontend/src/services/likeService.js
import { api } from "@/services/api"

export const likeTweet = async (tweetId) => {
    const response = await api.post("/like", {
      tweetId,
    });
  
    return response.data;
  };

  export const dislikeTweet = async (tweetId) => {
    const response = await api.post("/dislike", {
      tweetId,
    });
  
    return response.data;
  };